### PR TITLE
Sideset moment postprocessor

### LINKED ISSUE #358: SidesetMoment postprocessor
## Reason
To compute the moment about a point given the stress tensor and the required direction.

## Design
SidesetMoment postprocessor class will be implemented.

## Impact
No impact on existing classes.

### PR DESCRIPTION
**Pull request by somu15**
## Reason
To compute the moment about a point given the stress tensor or pressure and the required direction.

## Design
SidesetMoment postprocessor class will be implemented.

## Impact
No impact on existing classes.

closes #358

### CODE REVIEW

**Comment by somu15** [comment]
@cbolisetti I made changes per the discussion we had last week.

**Comment by cbolisetti** [comment]
@somu15 we discussed adding more guidance on the stress direction, leverarm direction, and the direction of the resulting moment. Please add those too.

File: src/postprocessors/SidesetMoment.C [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+template <bool is_ad>
+Real
+SidesetMomentTempl<is_ad>::computeQpIntegral()
+{
+  if (_tensor)
+    return _normals[_qp] * (MetaPhysicL::raw_value((*_tensor)[_qp]) * (*_stress_dir)) *
+           (_q_point[_qp] - _ref_point) * (*_leverarm_direction);
+  else
+    return ((*_p)[_qp]) * (_q_point[_qp] - _ref_point) * (*_leverarm_direction);
```
Comment (cbolisetti) [inline src/postprocessors/SidesetMoment.C:67]: Is this valid only if the sideset normal is perpendicular to the leverarm direction?

File: src/postprocessors/SidesetMoment.C [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+template <bool is_ad>
+InputParameters
+SidesetMomentTempl<is_ad>::validParams()
+{
+  InputParameters params = SideIntegralPostprocessor::validParams();
+  params.addClassDescription("Computes the product of integrated reaction force and lever arm in a "
+                             "user-specified direction "
+                             "on a sideset from the surface traction");
+  params.addParam<MaterialPropertyName>("stress_tensor", "The rank two stress tensor name");
+  params.addParam<RealVectorValue>("stress_dir", "Stress direction");
```
Comment (cbolisetti) [inline src/postprocessors/SidesetMoment.C:25]: Change to "stress_direction" to be consistent with "leverarm_direction".

**Review by cbolisetti — CHANGES_REQUESTED** [review]

File: doc/content/source/postprocessors/SidesetMoment.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+\end{equation}
+
+The figure below illustrates `SidesetMoment` for a simple example. Moment is to be
+computed about the z-axis (out-of-plane) at the reference point using either the
+stress tensor or the pressure at the right boundary. If the stress tensor is used,
+the normal vector to the boundary is $\vec{n} = (1, 0, 0)$ and the desired stress
+direction is along x-axis (i.e., $\vec{e} = (1, 0, 0)$). The quantity $\big(\vec{n}~\cdot~\underline{\sigma}~\vec{e}\big) \vec{n}$ is a force-like vector. If the pressure is used, no operations on $p$ are necessary
+ since it is a scalar, and $p \vec{n}$ is the force-like vector. $(\vec{x}-\vec{r})$ is the
+ relative distance vector between the the current quadrature and reference points, and dotting
+ this with the moment direction vector $(\vec{l})$ gives the scalar lever arm.
```
Comment (cbolisetti) [inline doc/content/source/postprocessors/SidesetMoment.md:35]: Check your math and rewrite. This is not what you are doing above.

File: src/postprocessors/SidesetMoment.C [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+InputParameters
+SidesetMomentTempl<is_ad>::validParams()
+{
+  InputParameters params = SideIntegralPostprocessor::validParams();
+  params.addClassDescription("Computes the product of integrated reaction force and lever arm in a "
+                             "user-specified direction "
+                             "on a sideset from the surface traction");
+  params.addParam<MaterialPropertyName>("stress_tensor", "The rank two stress tensor name");
+  params.addParam<RealVectorValue>("stress_direction", "Stress direction");
+  params.addCoupledVar("p", "The scalar pressure");
```
Comment (cbolisetti) [inline src/postprocessors/SidesetMoment.C:26]: Change this to "pressure".

**Review by cbolisetti — CHANGES_REQUESTED** [review]

**Comment by somu15** [comment]
I've updated the figure, but for some reason, it's not showing the updated figure on the MOOSE test site.

**Review by cbolisetti — APPROVED** [review]

**Pull request merged**